core/vm/contracts.go

```diff
@@ -533,3 +544,59 @@ func (c *vrf) Run(input []byte) ([]byte, error) {
 	// So here we simply return it
 	return append([]byte{}, input...), nil
 }
+
+// SHA3-256 FIPS 202 standard implemented as a native contract.
+
+type sha3fip struct{}
+
+// TODO Check if the gas price calculation needs modification
+// RequiredGas returns the gas required to execute the pre-compiled contract.
+//
+// This method does not require any overflow checking as the input size gas costs
+// required for anything significant is so high it's impossible to pay for.
+func (c *sha3fip) RequiredGas(input []byte) uint64 {
+	return uint64(len(input)+31)/32*params.Sha256PerWordGas + params.Sha256BaseGas
```

Comment:
How is this gas formula determined?

Reply:
@rlan35 the gas price formula is kept the same as that of sha256hash precompiles. Thanks for pointing it out, I did a bit more in-depth research on the gas prices of various hashing functions, since SHA3-256 is the same family as keccak256.
the gas price should be of that as mentioned in the Ethereum yellow paper (appendix G)
![image](https://user-images.githubusercontent.com/5087697/129591847-7b7a5f53-3514-4a63-b5bb-fa4a462019d1.png).
we will change this to the values as shown above, agree?

Reply:
hi, @rlan35 , waiting for your goahead for the above as well as the epoch numbers as per previous conversation.

Reply:
Yes, 30 and 6 seems more reasonable here.

Reply:
Hi @rlan35 , Done with these changes, please review it